test: Expand mobile editor tests

dcalhoun opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
Expand test coverage for the mobile editor.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Reduce the likelihood of regressions.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
**test: Test multiple text entry editor history**
Expand editor history tests to cover entering text over multiple history
entries.

**test: List split and merge**
Expand tests to cover splitting lists and merging with paragraphs.

**test: Highlight text with selection**
Expand tests to cover highlighting text with selection.

**test: Update inline comments**
The to-do note is not possible for formatted strings that result in
inner HTML tags within the text input. This is because the
`typeInRichText` helper does not support parsing inner HTML tags, only
outer tags. This results in inaccurate cursor placement as the tags
occupy indexes within the string, so the target index is off given it is
based upon the visual position, i.e. the position in the rendered
version of the parsed HTML, rather than the literal position in the
string, i.e. taking into account what positions are occupied by HTML
tags.

If we want to test typing after applying formats, we should likely use
e2e tests.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->
n/a

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->
n/a

## Screenshots or screencast <!-- if applicable -->
n/a

dcalhoun committed:
test: Test multiple text entry editor history

Expand editor history tests to cover entering text over multiple history
entries.

dcalhoun committed:
test: List split and merge

Expand tests to cover splitting lists and merging with paragraphs.

dcalhoun committed:
test: Highlight text with selection

dcalhoun committed:
test: Update inline comments

The to-do note is not possible for formatted strings that result in
inner HTML tags within the text input. This is because the
`typeInRichText` helper does not support parsing inner HTML tags, only
outer tags. This results in inaccurate cursor placement as the tags
occupy indexes within the string, so the target index is off given it is
based upon the visual position, i.e. the position in the rendered
version of the parsed HTML, rather than the literal position in the
string, i.e. taking into account what positions are occupied by HTML
tags.

If we want to test typing after applying formats, we should likely use
e2e tests.

fluiddot commented on packages/block-library/src/list/test/edit.native.js:
```diff
@@ -338,52 +338,148 @@ describe( 'List block', () => {
 		fireEvent.press( listBlock );
 		await triggerBlockListLayout( listBlock );
+		const listItemField = screen.getByLabelText( /Text input. .*One.*/ );
+		selectRangeInRichText( listItemField, 3, 3 );
```
Nitpick: The `end` argument uses the `start` value by default, hence we don't need to pass it.
```suggestion
		selectRangeInRichText( listItemField, 3 );
```

https://github.com/WordPress/gutenberg/blob/f8e071ab10c9ff75788a654a4fd9112ba5235ab5/test/native/integration-test-helpers/rich-text-select-range.js#L14

fluiddot reviewed (approved):
LGMT 🎊 ! Awesome work on expanding the integration tests @dcalhoun 🥇 !

dcalhoun committed:
test: Remove redundant selectRangeInRichText argument

The helper defaults to passing the second parameter (`start`) as the third parameter (`end`).

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

dcalhoun committed:
test: Fix editor history test compatibility with new UI

The undo and redo buttons relocated out of the editor and into the host
app in c428fd43302c38a8d0c945452c41235e64bd1607.

dcalhoun committed:
test: Wrap undo and redo actions in act to avoid warnings

The undo and redo actions result in state changes for the editor, which
lead to `act` warnings in the tests. This wraps the actions in `act` so
that warnings are suppressed.

These new undo and redo actions were introduced in
c428fd43302c38a8d0c945452c41235e64bd1607.
